Dockerfile-ngc

```diff
@@ -0,0 +1,201 @@
+ARG HOROVOD_WITH_PYTORCH=1
+ARG HOROVOD_WITHOUT_MXNET=1
+ARG HOROVOD_WITH_MPI=0
+ARG HOROVOD_CPU_OPERATIONS=GLOO
+ARG HOROVOD_GPU_OPERATIONS
+ARG HOROVOD_WITHOUT_MPI=1
+#RUN pip install cmake==3.22.4 protobuf==3.20.3
+ENV HOROVOD_MPI=/container/ompi
+ENV HOROVOD_WITHOUT_MPI=1
+ENV HOROVOD_GPU_OPERATIONS=NCCL
+ENV HOROVOD_CPU_OPERATIONS=GLOO
+ENV HOROVOD_WITH_TENSORFLOW=0
+ENV HOROVOD_WITH_PYTORCH=1
+#ENV HOROVOD_BUILD_ARCH_FLAGS="-std=c++-17"
+#ENV CXXFLAGS="${CXXFLAGS} -std=c++-17"
+#RUN pip install "$HOROVOD_PIP" 
+#RUN if [ -n "${HOROVOD_PIP}" ]; then ldconfig /usr/local/cuda/targets/x86_64-linux/lib/stubs && \
+#    pip install "$HOROVOD_PIP" && \
+#    ldconfig; fi
+
+#ENV NCCL_LIB_DIR=${HOROVOD_NCCL_HOME}/lib
+#ENV LD_LIBRARY_PATH=${WITH_OFI:+$NCCL_LIB_DIR:}$LD_LIBRARY_PATH
+ARG AWS_PLUGIN_INSTALL_DIR=/container/aws
+ARG WITH_AWS_TRACE
+ARG INTERNAL_AWS_DS
+ARG INTERNAL_AWS_PATH
+RUN if [ "$WITH_OFI" = "1" ]; then /tmp/det_dockerfile_scripts/build_aws.sh "$WITH_OFI" "$WITH_AWS_TRACE"; fi
+#USING OFI
+ARG AWS_LIB_DIR=${AWS_PLUGIN_INSTALL_DIR}/lib
+ENV LD_LIBRARY_PATH=${WITH_OFI:+$AWS_LIB_DIR:}$LD_LIBRARY_PATH
+
+
+
+
+
+# We uninstall these packages after installing. This ensures that we can
+# successfully install these packages into containers running as non-root.
+# `pip` does not uninstall dependencies, so we still have all the dependencies
+# installed.
+RUN python -m pip install determined && python -m pip uninstall -y determined
+
+RUN python -m pip install -r /tmp/det_dockerfile_scripts/notebook-requirements.txt
+ENV JUPYTER_CONFIG_DIR=/run/determined/jupyter/config
+ENV JUPYTER_DATA_DIR=/run/determined/jupyter/data
+ENV JUPYTER_RUNTIME_DIR=/run/determined/jupyter/runtime
+
+RUN /tmp/det_dockerfile_scripts/add_det_nobody_user.sh
+
+RUN /tmp/det_dockerfile_scripts/install_libnss_determined.sh
+
+# Set an entrypoint that can scrape up the host libfabric.so and then 
+# run the user command. This is intended to enable performant execution
+# on non-IB systems that have a proprietary libfabric.
+RUN mkdir -p /container/bin && \
+    cp /tmp/det_dockerfile_scripts/scrape_libs.sh /container/bin
+ENTRYPOINT ["/container/bin/scrape_libs.sh"]
+CMD ["/bin/bash"]
+
+RUN /tmp/det_dockerfile_scripts/install_google_cloud_sdk.sh
+
+RUN python -m pip install -r /tmp/det_dockerfile_scripts/additional-requirements.txt
+
+#ARG DEEPSPEED_PIP
+#RUN if [ -n "$DEEPSPEED_PIP" ]; then /tmp/det_dockerfile_scripts/install_deepspeed.sh; fi
+
+# Copy Liam's permissions setting code from Lore project
+# Make sure permissions are okay for nonroot and cleanup.
+RUN chmod -R 777 /usr/local/lib/python3.10/dist-packages/smart* && chmod -R 777 /tmp
+# LL_NOTE: need to figure out why nvidia sets user to 1000 for the ngc container
+# that we are building from and whether there are any issues to us not doing the same
+# when building this container.
+RUN chown root /usr/lib && chgrp root /usr/lib
+
+# Force it to use a precompiled version of the aws ofi plugin

```

Comment:
This should be conditional because I think it only seems to matter if we are targeting the H100s, and possibly more specifically, the GH nodes. Perhaps a build arg that can be set when targeting GH.